pyiceberg/manifest.py

```diff
@@ -762,10 +762,10 @@ def add_entry(self, entry: ManifestEntry) -> ManifestWriter:
 
         if (
             (entry.status == ManifestEntryStatus.ADDED or entry.status == ManifestEntryStatus.EXISTING)
-            and entry.data_sequence_number is not None
-            and (self._min_data_sequence_number is None or entry.data_sequence_number < self._min_data_sequence_number)
+            and entry.sequence_number is not None
+            and (self._min_data_sequence_number is None or entry.sequence_number < self._min_data_sequence_number)
```

Comment:
nit: there are a few variables named `min_data_sequence_number`, should we change those to match? 
https://github.com/search?q=repo%3Aapache%2Ficeberg-python%20data_sequence_number&type=code

Reply:
That thought did cross my mind and then I checked the manifest list files and in the actual files it is correct (i.e. `sequence_number` and `min_sequence_number`). 
We only have (private) variable `_min_data_sequence_number` in `ManifestWriter` class, so I left it as it is. 
I guess we should standardize them and make the variable names same as fields in spec.

Reply:
i think standardizing them would make it easier to associate the two variables in the future

Reply:
Renamed _min_data_sequence_number to _min_sequence_number in ManifestWriter.

Reply:
Also renamed `min_data_sequence_number` to `min_sequence_number` in init file pyiceberg.table package, no change required in tests as the variable usage was internal.